src/client/Microsoft.Identity.Client/AccountExtensions.cs

```diff
@@ -23,5 +23,14 @@ public static IEnumerable<TenantProfile> GetTenantProfiles(this IAccount account
         {
             return (account as Account)?.TenantProfiles;
         }
+
+        /// <summary>
+        /// Returns the authority from which the account is acquired.
+        /// </summary>
+        /// <returns>Authority for the account.</returns>
+        public static string GetAuthority(this IAccount account)
+        {
+            return $"https://{account.Environment}/{account.HomeAccountId.TenantId}";
```

Comment:
Does it work in the case of B2C ? (I would not think so)

Reply:
Yes, I had the same comment. We need to have unit tests for this for ADFS and B2C